src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -51,6 +52,7 @@ internal class SqlServerSearchService : SearchService
         private readonly SchemaInformation _schemaInformation;
         private readonly ISortingValidator _sortingValidator;
         private readonly IFhirRequestContextAccessor _requestContextAccessor;
+        private const int _resourceTableColumnCount = 11;
```

Comment:
>_resourceTableColumnCount [](start = 26, length = 25)

I count it twice and resource table has 10 fields!

Reply:
😵 interesting, good catch. @itye-msft, do you know why we compare against one greater than `reader.FieldCount`?

I set `_resourceTableColumnCount = 10` and used `_resourceTableColumnCount + 1` in 620e165.

Reply:
I'll try answering the question. For non-sort queries, the final result set from the SQL query contains 10 fields (I think it's just a co-incidence that it's equal to the number of columns in the `dbo.ResourceTable` - since the 10 also includes IsMatch and IsPartial). For sort queries, the final result contains one more field since we also extract the `SortValue`. Hence the check.

Reply:
nit: I think it's a good idea to give a name to the "magic number" 10. But I feel that `_resourceTableColumnCount` might not be right since that is not what we are actually referring to here. Maybe something like `resultSetColumnCount`? Even if we do rename it, we should do it as part of a different PR and not this one.